packages/hub/routes/utils/orders.ts

```diff
@@ -0,0 +1,181 @@
+import { Client as DBClient } from 'pg';
+import Web3 from 'web3';
+import RelayService from '../../services/relay';
+import Subgraph from '../../services/subgraph';
+
+const { toChecksumAddress } = Web3.utils;
+
+export type OrderStatus =
+  | 'waiting-for-order'
+  | 'received-order'
+  | 'waiting-for-reservation'
+  | 'provisioning'
+  | 'error-provisioning'
+  | 'complete';
+export type OrderEvent =
+  | 'received-reservation'
+  | 'wyre-receive-funds'
+  | 'wyre-send-funds'
+  | 'provision-error'
+  | 'provision-mined';
+export interface OrderState {
+  reservationId: string | null;
+}
+
+/* State machine for order status:
+
+     {START}
+       |  \
+       |   \  received-reservation
+       |    \
+       |     V
+       |    {WAITING-FOR-ORDER}
+wyre   |           \
+receive|            \  wyre-receive-funds
+funds  |             \         ______
+       |              V       /      \  received
+       |------> {RECEIVED-ORDER}      | reservation
+                   /       |  ^______/
+                  /        |
+wyre-send-funds  /         |
+reservationId == null      |
+               /           |
+              V            | wyre-send-funds
+{WAITING-FOR-RESERVATION}  | reservationId != null
+              \            |
+  received     \           |
+  reservation   \          |
+                 V         V
+                {PROVISIONING}
+                     /    \
+  provisioned-mined /      \ provision-error
+                   /        \
+                  V          V
+             {COMPLETE}    {ERROR-PROVISIONING}
+
+*/
```

Comment:
This state machine is important for understanding how prepaid cards are provisioned

Reply:
(amazingly github co-pilot did most of the drawing here)